Lib/pprint.py

```diff
@@ -525,6 +527,13 @@ def _safe_repr(self, object, context, maxlevels, level):
             return repr(object), True, False
 
         r = getattr(typ, "__repr__", None)
+
+        if issubclass(typ, int):
+            if self._underscore_numbers:
+                return "{:_d}".format(object), True, False
```

Comment:
I assume there's a conflict with another function named `format`, but using the builtin `format` function would be preferable. If not that, then I'd suggest an f-string over `str.format`. Either the builtin `format` or a f-string would have better performance. Not that it probably matters very much.

Reply:
I changed the string format to `builtins.format()`. It works with the explicit namespace. I think it fixes the issue you say with minimal changes.
Is it ok for you?

Reply:
According to [this message](https://github.com/python/cpython/pull/24864#discussion_r598348655), a f-string is now used to format the integer.
I close this conversation because there is another thread involving gpshead about the same topic.